src/hashes/ternary/curl_p/regular/transform.rs

```diff
@@ -23,22 +19,32 @@ struct StateRotation {
     shift: u8,
 }
 
-lazy_static! {
-    static ref STATE_ROTATIONS: [StateRotation; NUM_ROUNDS] = {
-        let mut rotation = ROTATION_OFFSET;
+const STATE_ROTATIONS: [StateRotation; NUM_ROUNDS] = {
+    let mut rotation = ROTATION_OFFSET;
 
-        let mut state_rotations = [StateRotation { offset: 0, shift: 0 }; NUM_ROUNDS];
+    let mut state_rotations = [StateRotation { offset: 0, shift: 0 }; NUM_ROUNDS];
 
-        for state_rotation in &mut state_rotations {
-            state_rotation.offset = rotation / HASH_LENGTH;
-            // This is fine since `HASH_LENGTH <= u8::MAX`.
-            state_rotation.shift = (rotation % HASH_LENGTH).try_into().unwrap();
-            rotation = (rotation * ROTATION_OFFSET) % STATE_SIZE;
-        }
+    let mut i = 0;
 
-        state_rotations
-    };
-}
+    while i < NUM_ROUNDS {
+        state_rotations[i].offset = rotation / HASH_LENGTH;
+        state_rotations[i].shift = {
+            let shift = rotation % HASH_LENGTH;
+
+            if shift > 255 {
+                panic!("shift is too large");
```

Comment:
`HASH_LENGTH = 243` , this branch is impossible.

Reply:
Yeah I know, that's why this is done during compilation so if someone changes the code logic they actually notice that they messed up.